stl/inc/ranges

```diff
@@ -2040,7 +2040,7 @@ namespace ranges {
             template <class _Rng>
             static constexpr _Choice_t<_St> _Choice = _Choose<_Rng>();
 
-            template <_Integer_like _Ty>
+            template <class _Ty>
             struct _Partial : _Pipe::_Base<_Partial<_Ty>> {

```

Comment:
The constraint below does not appear to match the conversion actually applied.

I also note that WG21-P2281 requires perfect forwarding call wrapper semantics, which this isn't. Admittedly, it's hard to think of a non-contrived example where that makes a difference.

Reply:
Is the issue that the move construction is written with copy-initialization instead of direct-initialization syntax, or something else? Can you suggest a better constraint?

Reply:
I was referring to the `convertible_to<_Ty&, range_difference_t<_Rng>>` on line 2049. `_Length` is `const` inside a `const` member function.

Reply:
Spammed out all four flavors of `operator()` with proper call-forwarding behavior and constraints, for both `_{Drop,Take}_fn::_Partial::operator()`.